src/main/java/at/ac/tuwien/kr/alpha/grounder/heuristics/GrounderHeuristicsConfiguration.java

```diff
@@ -0,0 +1,150 @@
+ * 
+ * 2) Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package at.ac.tuwien.kr.alpha.grounder.heuristics;
+
+import at.ac.tuwien.kr.alpha.grounder.Grounder;
+
+/**
+ * Contains configuration parameters for heuristics used by {@link Grounder}s.
+ * 
+ * Both parameters {@link #toleranceConstraints} and {@link #toleranceRules} are interpreted as follows:
+ * A rule (or constraint) is grounded if the following conditions are satisfied:
+ * <ul>
+ * 	<li>All variables in the rule are bound (either by positive body literals that are already
+ *      satisfied, or because the whole rule is ground).</li>
+ * 	<li>No atom occurring positively in the body is assigned F.</li>
+ * 	<li>No atom occurring negatively in the body is true as a fact.
+ * 		(Note: The rule is still grounded if an atom occurring negatively in the body is assigned true without
+ * 		being a fact, because the alternative would necessitate triggering re-grounding during backtracking.)</li>
+ * 	<li>At most {@code N} atoms occurring positively in the body are still unassigned.</li>
+ * </ul>
+ * 
+ * The parameter {@link #toleranceConstraints} specifies {@code N} for constraints, while {@link #toleranceRules}
+ * specifies {@code N} for all other rules. Infinity is represented by the value {@code -1}.
+ * The default value for both parameters is {@code 0}, which means that only those rules and constraints are
+ * grounded whose positive body is already satisfied.
+ *
+ * The additional parameter {@link #disableInstanceRemoval} is a switch for the accumulator grounding strategy
+ * which disables the removal of instances from the grounder memory in certain cases.
+ *
+ */
+public class GrounderHeuristicsConfiguration {
+
+	public static final String STRICT_STRING = "strict";
+	public static final int STRICT_INT = 0;
+	public static final String LAX_STRING = "lax";
+	public static final int LAX_INT = -1;
+	
+	private int toleranceConstraints;
+	private int toleranceRules;
+	private boolean disableInstanceRemoval;
```

Comment:
I wonder whether we should rename this, to something like `accumulationEnabled`?

Reply:
Agreed. In 6b8dd0bbf12f156f545e83e8a4a0a7f19e6e4845 I also changed names from "lax" to "permissive" to conform with published papers.